pkg/render/nonclusterhost/nonclusterhost.go

```diff
@@ -0,0 +1,212 @@
+// Copyright (c) 2024 Tigera, Inc. All rights reserved.
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package nonclusterhost
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	rbacv1 "k8s.io/api/rbac/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	operatorv1 "github.com/tigera/operator/api/v1"
+	"github.com/tigera/operator/pkg/common"
+	"github.com/tigera/operator/pkg/render"
+	rmeta "github.com/tigera/operator/pkg/render/common/meta"
+)
+
+const (
+	NonClusterHostObjectName = "tigera-noncluster-host"
+)
+
+type Config struct {
+	NonClusterHost operatorv1.NonClusterHostSpec
+}
+
+func NonClusterHost(cfg *Config) render.Component {
+	return &nonClusterHostComponent{
+		cfg: cfg,
+	}
+}
+
+type nonClusterHostComponent struct {
+	cfg *Config
+}
+
+func (c *nonClusterHostComponent) ResolveImages(is *operatorv1.ImageSet) error {
+	return nil
+}
+
+func (c *nonClusterHostComponent) SupportedOSType() rmeta.OSType {
+	return rmeta.OSTypeLinux
```

Comment:
I think this should be OSTypeAny; no pods are deployed as a result of this.

Reply:
Changed to `OSTypeAny`.